### DISCUSSION TITLE
Progress bar not updated when downloading maps

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#12] [Q&A] [answered]
The progress bar is not updated when downloading maps

**Comment by devemux86** [comment]
Thanks for the report. Can you write more details?

Do you mean the progress bar inside the app or in the notifications at the top?
Or the progress bar in notifications was updated, but not in the app?

The downloads are handled by Android and sometimes it take a while to start.

What map from which provider?
Did you start the download and wait by looking at the screen or close / reopen the downloader?

**Reply by DolomitesTom** [reply]
progress bar inside the app...  The bar in the notifications is updating correctly.... Mapsforge, no matter which map, all, in this case germany... if you look at it or if you close the screen and look later, it doesn't update the bar. At the end when the download is completed than the bar is updated fully in a second.

**Reply by devemux86** [reply]
The only time I can get this to happen is if exit the app, by double tapping the Android back button.
(I don't mean to put the app in the background, by tapping the Android home button, then it works)

When the app is in the foreground or background and the download screen is open or hidden (and reopen)
I can see the progress bar moving along with the bar in the notifications.

What do you see?
No progress bar at all or a bar that keeps going back and forth?

**Reply by 0709wiwiwi** [reply]
Correct progress bar on A9 phone. (Small map)
https://youtube.com/shorts/kyVxsyqEeX4

**Comment by devemux86** [comment]
> Correct progress bar on A9 phone.

- That's right, at first until the download starts, the bar goes back and forth
- Then download starts and the bar moves forward (same with notifications)
- And finally in the unzip and copy stage, the bar goes again back and forth

The downloads are handled by Android and should not exit from the app.
(may go into the background)

**Comment by WolfgangZ42** [comment]
On my A10 phone sometimes the progress bar is updated, and sometimes it is not. But that's not important for me - I get an acoustic signal if it's ready and the Android download shows the progress either. 
I really appreciate the (semi) automatic update of my downloaded maps:
![2023-01-07 16 53 21](https://user-images.githubusercontent.com/121806238/211191172-f7374726-d0d9-4858-8fff-4513ba80d7b6.png)
Thank you Emux for realizing that!

**Comment by DolomitesTom** [comment]
> What do you see?
> No progress bar at all or a bar that keeps going back and forth?

just tested on a Samsung A52S (other was a Samsung S9 plus): a bar that keeps going back and forth is shown for 2-3 seconds, then a blank bar is seen until the download is completed, then again for some seconds a bar that is going back and forth. It seems to be an issue with large maps only. A small map like luxemburg shows the progress bar, but it is downloaded within seconds. You cannot simulate it with such little maps.

It is not a big issue. You can leave it alone until the map has gone and is visible under local and then it is ok

**Reply by devemux86** [reply]
Thanks for the extra details.
So it is about the 2nd stage (as described [above](https://github.com/devemux86/cruiser/discussions/12#discussioncomment-4620779)) where the app progress bar should move together with the one in notifications.
So I need to check again what happens at that stage.

**Reply by 0709wiwiwi** [reply]
I see this as follows.
1. Download shows the normal Cruiser animation.
2. Due to Android time out, your screen goes black.
3. Reopening the screen the animation is than halted.

**Comment by devemux86** [comment]
> Due to Android time out, your screen goes black.
> Reopening the screen the animation is than halted.

Downloads are handled by Android (as in browsers).
For large downloads anything can happen, better watch them.

As downloads progress and you have not exited Cruiser
(you can put it in background with Android home button),
when you open Cruiser again and its download manager,
its progress bar syncs with the progress in notifications.

Anything different should be something I need to check. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
Works like this

**Comment by devemux86** [comment]
Next Cruiser version will include in the downloader the **Voluntary** themes: https://voluntary.nichesite.org

(working on this I improved the downloader, let's see if this report happens again)

<img src="https://user-images.githubusercontent.com/3484020/212484619-7e1c505f-0eec-4d45-9dc5-77ca6917921e.jpg" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
[Cruiser 3.0.8](https://play.google.com/store/apps/details?id=gr.talent.cruiser) is available with improvements in download manager.

Please see if everything is working properly now.

**Reply by WolfgangZ42** [reply]
In the last days I did some downloads (Spain and Portugal) with 3.0.7 and saw a correct progress bar ...

**Comment by MotoUKRider** [comment]
Just updated the France map (2.9G).  The Cruiser progressbar did no show any value until over halfway on the NotificationBar progress, which displayed correctly.

I wonder if this is a "2GB Barrier" thing as smaller downloads are fine.
(child of the BBC Micro (shift-break), MS-DOS 8.3 era, FAT16 etc!)

**Reply by devemux86** [reply]
Maybe Android synchronization goes off after such long time.
Did you keep the app in the foreground during the download?

> (child of the BBC Micro (shift-break), MS-DOS 8.3 era, FAT16 etc!)

:+1: 

Amstrad, Commodore, Spectrum, nice times! :joystick:

**Reply by MotoUKRider** [reply]
Watched the Cruiser app the whole time - no drawing of the progressbar but when pulling-down the notification bar, it was accurate.  Once that was over 50%, the Cruiser progressbar showed 100% but it obviously sat there until the thing had downloaded and merged into the existing file-system.  Odd!

Ah, C64 - I could never undstand why friends put up with the waiting for the tape to load :)

**Reply by devemux86** [reply]
> Just updated the France map (2.9G)

Update, not a new download, right?

**Reply by WolfgangZ42** [reply]
I just tried to update the France map - process bar stays totally empty. The process bar of the Android download was correct. It was a very, very, very slow download, with 25 mb bandwidth. Cruiser was set into navigation and therefore permanently in foreground while updating France. After let say 20 minutes the process was interrupted - no longer an Android download at all.

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the test.

> After let say 20 minutes the process was interrupted - no longer an Android download at all.

Do you mean, you stopped it or it stopped by other reason?

**Reply by WolfgangZ42** [reply]
It was not stop by myself - I was patient enough. I always checked the Android process at intervals of a few minutes, and at some point this process no longer existed.

**Reply by MotoUKRider** [reply]
> Update, not a new download, right?

Yes, updating the existing version.

**Reply by devemux86** [reply]
Thanks everyone for your reports. :+1: 

I will check the progress bar with large downloads / updates.

**Comment by Ciutnik** [comment]
I also experienced this today, the progress bar stopped moving while downloading a 1.6 GB map. Most likely it was due to a temporary drop in network performance, because after a few minutes everything returned to normal. But at the time it occurred to me that maybe a simple numeric data transfer counter located on the download card would be a useful clue here.

Edit: I mean instantaneous download speed counter, without the need for a reset or any other interaction.

**Comment by devemux86** [comment]
Next Cruiser will display the download percentage instead of the progress bar, let's see how it goes:

<img src="https://github.com/devemux86/cruiser/assets/3484020/2bc935c4-005d-4810-b683-39b46cfbf690" width="50%"/>

**Reply by devemux86** [reply]
> Next Cruiser will display the download percentage instead of the progress bar, let's see how it goes:

Available in [Cruiser 3.0.23](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
